packages/scheduler/lib/models/tasks.ts

```diff
@@ -305,3 +306,36 @@ export async function expiresIfTimeout(db: knex.Knex): Promise<Result<Task[]>> {
         return Err(new Error(`Error expiring tasks: ${stringifyError(err)}`));
     }
 }
+
+export async function hardDeleteOlderThanNDays(db: knex.Knex, days: number): Promise<Result<Task[]>> {
+    try {
+        // Delete terminated tasks where lastStateTransitionAt is older than N days
+        // unless it is the most recent task for an given schedule
+        const tasks = await db
+            .from<DbTask>(TASKS_TABLE)
+            .where(
+                'id',
+                '=',
+                // schedules.id is null if the task is not associated with a schedule (ie: actions)
+                // or if the task is not the last task for its associated schedule
```

Comment:
subquery below is a bit more complicated than the schedules one because we need to keep the latest tasks for a given schedule even if it is older than 30 days. This is achieved by joining with `schedules.last_scheduled_task_id`. Any row with an empty left join is either a task without schedule (ie: actions) or a task that is not the latest one for any schedule

Reply:
Do we really need to keep the latest tho? We use the latest to be able to dedup tasks but ultimately if it's 60 days old it's not running for sure

Reply:
In theory syncs can have a frequency greater than 30 days, which would break scheduling/reporting if we delete the last one. 
In practice, current max frequency is actually 30 days but I prefer to cover this case so we don't run into issues if customers set frequency > 30days or if we decide in the future to lower this 30 days arbitrary threshold